### PR TITLE
mouse cursor wip

### PR DESCRIPTION
**Pull request by yuvalab**

### CODE REVIEW

File: packages/lib/src/settings/options/index.js [hunk 1]
```diff
@@ -355,6 +356,7 @@ export default {
     layoutParams_navigationArrows_container_borderRadius,
   [optionsMap.stylingParams.temResolutionMode]:
     stylingParams_itemResolutionMode,
+  mouseCursorContainer,
 };
 // TODO = add the options:
```
Comment (yuvalab) [inline packages/lib/src/settings/options/index.js]: options.ts - inteface adding the option

Comment (yuvalab) [reply packages/lib/src/settings/options/index.js]: change and import the right name of the sp

File: packages/gallery/src/components/helpers/mouseCursorPosition.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { useEffect, useState, useRef } from 'react';
+
+const throttle = (func, time) => {
```
Comment (yhattav) [inline packages/gallery/src/components/helpers/mouseCursorPosition.js]: find and use utils.throttle. if it doesnt exist create it and use it. this is very generic

File: packages/gallery/src/components/helpers/mouseCursorPosition.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      if (x > 0 && y > 0 && mouseIn) {
+        setPosition({
+          x,
+          y,
+        });
+      }
+    });
+  }, []);
+
+  console.log(position);
```
Comment (yhattav) [inline packages/gallery/src/components/helpers/mouseCursorPosition.js]: remove log

File: packages/gallery/src/components/gallery/proGallery/slideshowView.js [hunk 4]
```diff
@@ -606,8 +612,10 @@ class SlideshowView extends React.Component {
       arrowsType: layoutParams.navigationArrows.type,
       containerStyleType: type,
     });
+    const mouseCursorEnabled = arrowsPosition === 2;
```
Comment (yhattav) [inline packages/gallery/src/components/gallery/proGallery/slideshowView.js]: use CONST. change the naming of the character.

File: packages/gallery/src/components/helpers/mouseCursorPosition.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+
+const throttle = (func, time) => {
```
Comment (yhattav) [inline packages/gallery/src/components/helpers/mouseCursorPosition.js]: use utils or create throttle in the utils to be commonly used

File: packages/gallery/src/components/helpers/renderCursorController.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+            isMouseEnter,
+          },
+          ...props,
+        })
+      }
+    </CursorController>
+  );
+}
+
+export function renderArrowButton({
```
Comment (yhattav) [inline packages/gallery/src/components/helpers/renderCursorController.js]: why is this related to cursor controller (i.e why in this file and not an arrow related file?

**Review by yhattav — CHANGES_REQUESTED** [review]

**Pull request closed without merging**